cachi2/core/package_managers/pip.py

```diff
@@ -109,6 +121,28 @@ def fetch_pip_source(request: Request) -> RequestOutput:
     )
 
 
+def _generate_hash_from_string(string: str) -> Hash:
```

Comment:
According to the [Pip docs](https://pip.pypa.io/en/stable/cli/pip_hash/), these are the only possible values: sha256, sha384 and sha512.